pkg/kubelet/cm/devicemanager/topology_hints_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package devicemanager
+
+import (
+	"reflect"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1beta1"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+)
+
+func TestGetTopologyHints(t *testing.T) {
+	firstSocketMask, _ := socketmask.NewSocketMask(0)
+	secondSocketMask, _ := socketmask.NewSocketMask(1)
+	crossSocketMask, _ := socketmask.NewSocketMask(0, 1)
+	tcases := []struct {
+		name          string
+		amount        int64
+		expectedHints []topologymanager.TopologyHint
+	}{
+		{
+			name:   "Socket Affinity include 0, 1, 0 and 1",
+			amount: 1,
+			expectedHints: []topologymanager.TopologyHint{
+				{
+					SocketAffinity: firstSocketMask,
+					Preferred:      true,
+				},
+				{
+					SocketAffinity: secondSocketMask,
+					Preferred:      true,
+				},
+				{
+					SocketAffinity: crossSocketMask,
+					Preferred:      false,
+				},
+			},
+		},
+		{
+			name:   "Socket Mask includes {1, 1}",
+			amount: 2,
+			expectedHints: []topologymanager.TopologyHint{
+				{
+					SocketAffinity: crossSocketMask,
+					Preferred:      false,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcases {
+		m := ManagerImpl{}
+
+		testPod := v1.Pod{}
+		testContainer := v1.Container{}
+
+		testDevices := make(map[string][]pluginapi.Device)
+		testDevice1 := pluginapi.Device{ID: "ID", Health: pluginapi.Healthy, Topology: &pluginapi.TopologyInfo{Socket: int64(0)}}
+		testDevice2 := pluginapi.Device{ID: "ID", Health: pluginapi.Healthy, Topology: &pluginapi.TopologyInfo{Socket: int64(1)}}
```

Comment:
This will need to change with the new change made in https://github.com/kubernetes/kubernetes/pull/74423

```
testDevice1 := pluginapi.Device{ID: "ID", Health: pluginapi.Healthy, Topology: &pluginapi.TopologyInfo{Node: &pluginapi.NUMANode{0}}}
testDevice2 := pluginapi.Device{ID: "ID", Health: pluginapi.Healthy, Topology: &pluginapi.TopologyInfo{Node: &pluginapi.NUMANode{1}}}
```

Reply:
Yep I need to update this PR to reflect those changes, I just wanted to wait until we got a consensus on #74423 before making those changes